MBP-330: Brake with diagnostic function

mac-kan opened the pull request.

mac-kan committed:
Add input and status bools for brake diagnostics

Adds an input word to facilitate mapping diagnostics from e.g. an
EL2044 terminal to monitor if the output fails.

Adds status variables for brake diagnostics.

mac-kan committed:
Use the brake terminal diagnostic input

Adds the diagnostic input signal to the external brake code to prevent
unwanted motion when the diagnostic input fails. Since the input
variable defaults to TRUE, it is still possible to use the external
brake functionality without connecting the diagnostic input variable to
any I/O.

federrg reviewed (changes requested):
Correct me if I am wrong. Currently, it is just a boolean output. Meaning that it can only be linked to one of the diagnostic channels of the EL2044.
The EL2044 has 

<img width="330" height="343" alt="Image" src="https://github.com/user-attachments/assets/21c90f85-cff2-4ef4-baa8-4e1d9b650fa7" />

IF this is meant to only be linked to the open load diagnostics, I would change the variable name to that. If it is meant to be linked to more inputs, then we need to do something about that in the code.
@AlSallami7 is open load enough?

mac-kan commented:
@federrg I think that is a good decision. I'll fix that asap.

mac-kan committed:
Mask brake terminal diagnostics word

Masks the brake terminal diagnostics word and maps to two AxisStatus
variables.

mac-kan committed:
Update warning note message

federrg committed:
Merge remote-tracking branch 'origin/master' into MBP-330-Brake-With-Diagnostic-Function
